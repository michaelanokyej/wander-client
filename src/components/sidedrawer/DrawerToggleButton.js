import React from 'react';
import './DrawerToggleButton.css';
import toursAndUserContext from "/Users/michael/Documents/Courses/Thinkful/projects/capstone_projects/Wander/wander-client/src/context/toursAndUserContext.js";


class DrawerToggleButton extends React.Component {
  static contextType = toursAndUserContext;

  render() {
    return (
      <button className="toggle-button" onClick={e => this.context.drawerToggleClickHandler()}>
    <div className="toggle-button__line"></div>
    <div className="toggle-button__line"></div>
    <div className="toggle-button__line"></div>
  </button>
    )
  }
}

export default DrawerToggleButton;
