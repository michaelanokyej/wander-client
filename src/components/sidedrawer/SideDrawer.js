import React from "react";
import { Link } from "react-router-dom";
import "./SideDrawer.css";
import toursAndUserContext from "../context/toursAndUserContext.js";

class SideDrawer extends React.Component {
  static contextType = toursAndUserContext;

  linkToProfile = `/users/${this.context.userName}`;


  render() {
    // console.log(this.props.show);
    let drawerClasses = "side-drawer";
    if (this.props.show) {
      drawerClasses = "side-drawer open";
    }

    return (
      <nav className={drawerClasses}>
        <ul>
          {this.context.loggedIn ? (
            <>
              <li onClick={() => {
                this.context.backDropClickHandler()
              }}>
                <Link to="/home">Home</Link>
              </li>
              <li onClick={() => {
                this.context.backDropClickHandler()
              }}>
                <Link to="/post-a-tour">Post Tour</Link>
              </li>

              <li onClick={() => {
                this.context.backDropClickHandler()
              }}>
                {" "}
                <Link to={this.linkToProfile}>
                  {this.context.userName || "Profile"}
                </Link>
              </li>
            </>
          ) : (
            <li>
              <Link to="/signup">Sign Up</Link>
            </li>
          )}
        </ul>
      </nav>
    );
  }
}

export default SideDrawer;
