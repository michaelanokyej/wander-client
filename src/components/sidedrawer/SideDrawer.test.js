import React from "react";
import SideDrawer from "./SideDrawer";
import DrawerToggleButton from "./DrawerToggleButton";

import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe(`SideDrawer component`, () => {
  it(`renders the SideDrawer component`, () => {
    const wrapper = shallow(<SideDrawer />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it(`renders the DrawerToggleButton component`, () => {
    const wrapper = shallow(<DrawerToggleButton />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
