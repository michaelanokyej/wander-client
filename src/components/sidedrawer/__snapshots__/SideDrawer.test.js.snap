// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideDrawer component renders the DrawerToggleButton component 1`] = `
<button
  className="toggle-button"
  onClick={[Function]}
>
  <div
    className="toggle-button__line"
  />
  <div
    className="toggle-button__line"
  />
  <div
    className="toggle-button__line"
  />
</button>
`;

exports[`SideDrawer component renders the SideDrawer component 1`] = `
<nav
  className="side-drawer"
>
  <ul>
    <li
      onClick={[Function]}
    >
      <Link
        to="/signup"
      >
        Sign Up
      </Link>
    </li>
  </ul>
</nav>
`;
